### PR TITLE
update basic auth middleware

### PR DESCRIPTION
**Pull request by dominiksipowicz**
### Description

- Update basic Auth middleware example for middleware GA

### Best Way to Test

Login credentials:

- Username: 4dmin
- Password: testpwd123

before preview link: https://edge-functions-basic-auth-password.vercel.app/
after preview link: https://edge-functions-basic-auth-password-git-sol-340-middleware-auth.vercel.sh/

### CODE REVIEW

**Review by okbel — APPROVED** [review]

File: edge-functions/basic-auth-password/pages/auth.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+export async function getServerSideProps({ req, res }) {
+  res.setHeader('WWW-authenticate', 'Basic realm="Secure Area"')
+  res.statusCode = 401
+
+  return {
+    props: {},
+  }
+}
```
Comment (lfades) [inline edge-functions/basic-auth-password/pages/auth.tsx:12]: This should be an API route instead.

File: edge-functions/basic-auth-password/middleware.ts [hunk 2]
```diff
@@ -1,21 +1,24 @@
+/* eslint-disable @next/next/no-server-import-in-page */
```
Comment (lfades) [inline edge-functions/basic-auth-password/middleware.ts:1]: Install the `canary` version of `eslint-config-next` and you should be able to remove this line.

**Review by lfades — COMMENTED** [review]
Added some comments!

**Comment by lfades** [comment]
The example should have an `.npmrc` or `npm install` will fail locally and on Vercel.

**Pull request merged**